src/Streamly/Internal/Data/Producer/Source.hs

```diff
@@ -0,0 +1,91 @@
+-- |
+-- Module      : Streamly.Internal.Data.Producer.Source
+-- Copyright   : (c) 2021 Composewell Technologies
+-- License     : BSD-3-Clause
+-- Maintainer  : streamly@composewell.com
+-- Stability   : experimental
+-- Portability : GHC
+--
+-- A stream source which can be unfolded with some possibly buffered data.
+-- Allows to 'unread' data i.e. push unused data back to the source. This is
+-- useful in parsing applications with backtracking.
+--
+
+module Streamly.Internal.Data.Producer.Source
+    ( Source
+
+    -- * Creation
+    , source
+
+    -- * Transformation
+    , unread
+
+    -- * Consumption
+    , isEmpty
+    , producer
+    )
+where
+
+#include "inline.hs"
+
+import Streamly.Internal.Data.Stream.StreamD.Step (Step(..))
+import Streamly.Internal.Data.Producer.Type (Producer(..))
+import Prelude hiding (read)
+
+-- XXX instead of a seed should we wrap a Producer in a source? Then we won't
+-- have to supply a Producer for read.
+--
+-- | An unfold seed with some buffered data. It allows us to 'unread' or return
```

Comment:
unfold seed? producer seed?

Reply:
Chnaged all such occurrences in Source.hs.